### DISCUSSION TITLE
Where was getQueryParams moved to?

### DISCUSSION THREAD

**Discussion by iemong** [storybookjs/storybook#26491] [Help] [answered]
### Summary

> @storybook/client-api - this package has been merged into 

ref: https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#removed-deprecated-shim-packages

According to the migration docs, `@storybook/client-api` seems to have been merged to `@storybook/preview-api`, but the `getQueryParams` that `@storybook/client-api` provided seems to be missing.

The version of `@storybook/preview-api` is 8.0.0

**Comment by valentinpalkovic** [comment, marked as answer]
Hi @iemong 

Thank you for reporting this. As far as I can see, the API was removed together with the StoryStoreV6 support in this PR:
https://github.com/storybookjs/storybook/pull/24658.

Specifically, this commit removed the export: https://github.com/storybookjs/storybook/commit/534e49cfd0ba4c8bb778249638b92c6f49e70ef7#diff-69ccbe0dc95b01f15bf0cabf620ae7336ffcd54488d9048d4445b43449eb311f

Can you tell us in what specific use cases you rely on `getQueryParams`? I don't think we will bring this back, but the previous implementation of `getQueryParams` was pretty straightforward if you want to implement it by yourself:

```tsx
import { parse } from 'qs';

export const getQueryParams = () => {
  const { document } = global;
  // document.location is not defined in react-native
  if (document && document.location && document.location.search) {
    return parse(document.location.search, { ignoreQueryPrefix: true });
  }
  return {};
};
```

**Reply by iemong** [reply]
Thank you for your help @valentinpalkovic 

I got it. I'll implement `getQueryParams` by myself
Actually, we implemented a custom switcher of storybook UI for i18n and the switcher can dynamically change the query parameter of a URL.  ex) `http://localhost:6006/?path=/story/components-article--default&providerSwitcher-locale=ja-JP`
we use `getQueryParams` in that function
